File: backend/internal/domain/models/media.go
package models

type Media struct {
	BaseModel
	PlaylistID *string `db:"playlist_id" json:"playlist_id"`
	Name       string  `db:"name" json:"name"`
	Poster     *string `db:"poster" json:"poster"`
	Format     string  `db:"format" json:"format"`
	Path       string  `db:"path" json:"path"`
	Duration   int     `db:"duration" json:"duration"`
	Size       int     `db:"size" json:"size"`
	Order      int     `db:"order" json:"order"`
}

type Playlist struct {
	BaseModel
	Name   string  `db:"name" json:"name"`
	Type   string  `db:"type" json:"type"`
	Poster *string `db:"poster" json:"poster"`
}

File: frontend/src/api/media.ts
import apiClient from './client'
import type {
  Media,
  CreateMediaRequest,
  UpdateMediaRequest,
  PaginatedResponse,
  ApiResponse,
  BrowseMediaFilesResponse
} from '@/types/api'

export const mediaApi = {
  getMedia: async (params?: {
    page?: number
    limit?: number
    offset?: number
    name?: string
    format?: string
    only_unassigned?: boolean
  }): Promise<PaginatedResponse<Media>> => {
    const response = await apiClient.get<PaginatedResponse<Media>>('/media/', { params })
    return response.data
  },

  getMediaById: async (id: string): Promise<Media> => {
    const response = await apiClient.get<ApiResponse<Media>>(`/media/${id}/`)
    return response.data.result
  },

  createMedia: async (data: CreateMediaRequest): Promise<Media> => {
    const response = await apiClient.post<ApiResponse<Media>>('/media/', data)
    return response.data.result
  },

  updateMedia: async (id: string, data: UpdateMediaRequest): Promise<Media> => {
    const response = await apiClient.patch<ApiResponse<Media>>(`/media/${id}/`, data)
    return response.data.result
  },

  deleteMedia: async (id: string): Promise<void> => {
    await apiClient.delete(`/media/${id}/`)
  },

  uploadPoster: async (file: File): Promise<{ filename: string }> => {
    const formData = new FormData()
    formData.append('file', file)

    const response = await apiClient.post<ApiResponse<{ filename: string }>>('/upload/poster/', formData, {
      headers: {
        'Content-Type': 'multipart/form-data'
      }
    })
    return response.data.result
  },

  browseMediaFiles: async (path: string = ''): Promise<ApiResponse<BrowseMediaFilesResponse>> => {
    const response = await apiClient.get<ApiResponse<BrowseMediaFilesResponse>>('/media/browse', {
      params: { path }
    })
    return response.data
  }
}